src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -1342,7 +1342,7 @@ private ValueTask WaitForDataAsync(CancellationToken cancellationToken)
                     {
                         // Wake up the wait.  It will then immediately check whether cancellation was requested and throw if it was.
                         thisRef._waitSource.SetException(ExceptionDispatchInfo.SetCurrentStackTrace(
-                            CancellationHelper.CreateOperationCanceledException(null, thisRef._waitSourceCancellation.Token)));
+                            CancellationHelper.CreateOperationCanceledException(null, cancellationToken)));
```

Comment:
I believe you can now delete the CancellationHelper.ThrowIfCancellationRequested below.

Reply:
Yup, missed that.  Deleted.